### ISSUE TITLE
Python3.9: e.g. aiohttp TypeError: GenericAlias expected 2 arguments, got 3

### ISSUE THREAD

**Issue by chwba** [Nuitka/Nuitka#966]
After compiling and starting the excecutable via CMD, we get:
```
Traceback (most recent call last):
  File "C:\Users\ADMINI~1\PYCHAR~1\NUITKA~2\APP~1.DIS\app.py", line 4, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\ADMINI~1\PYCHAR~1\NUITKA~2\APP~1.DIS\aiohttp\__init__.py", line 6, in <module aiohttp>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\ADMINI~1\PYCHAR~1\NUITKA~2\APP~1.DIS\aiohttp\client.py", line 1121, in <module aiohttp.client>
TypeError: GenericAlias expected 2 arguments, got 3
```

Compiled with:
```
python -m nuitka --follow-imports^
                 --show-progress^
                 --show-scons^
                 --assume-yes-for-downloads^
                 --warn-unusual-code^
                 --plugin-enable=pylint-warnings^
                 --windows-dependency-tool=pefile^
                 --experimental=use_pefile_recurse^
                 --experimental=use_pefile_fullrecurse^
                 --standalone^
                 --mingw64^
                 app.py
```
Log: https://nextcloud.mnwd.in/s/op4xWKpqEkYTTkZ

**Used requirements:**
```
aiohttp
```

**Code snippet to reproduce:**
```python
import asyncio
from asyncio import sleep

import aiohttp


async def main():
	session = aiohttp.ClientSession()
	async with session.request(method='GET', url='https://google.com') as response:
		response_text = await response.text()
		print(response.status)
		print()
		print(response_text)
	await session.close()
	await sleep(1)

if __name__ == "__main__":
	asyncio.run(main())
```

**Nuitka**
```
0.6.11.3
Python: 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)]
Executable: C:\Users\Administrator\.pyenv\pyenv-win\versions\3.9.1\python.exe
OS: Windows
Arch: x86_64
```
Installed via `python -m pip install nuitka`

**Comment by kayhayen** [comment]
I didn't manage to reproduce this was with 3.8, trying 3.9 now.

**Comment by kayhayen** [comment]
And yes, that one shows it. So this seems to be something incompatible with 3.9 here.

**Comment by kayhayen** [comment]
Nuitka generates identical C codes for 3.8 and 3.9 in both cases, which is very welcome to see, but also not very revealing. :)

**Comment by kayhayen** [comment]
The minimal example I just distilled is this:

```
from typing import (
    Any,
    Coroutine,
)

class ClientResponse(object):
    pass

class _BaseRequestContextManager(Coroutine[Any, Any, Any]):
    pass

class _RequestContextManager(_BaseRequestContextManager[ClientResponse]):
    pass
```

**Comment by chwba** [comment]
Thank you for the quick responses :). I will be back home in one hour then I will start testing as well.

**Comment by kayhayen** [comment]
Well, the changelog is not normally on the level where Nuitka interacts with it.

But there definitely is a change because adding `GenericAlias` to the `types` module is new in CPython 3.9 and one thing, that Nuitka clearly doesn't know about, is how to call meta classes with only 2 arguments. Normally you pass name, bases, and dictionary. So even if my diffing of the CPython 3.9.1 vs. CPython 3.8.5 code didn't immediately reveal anything, something must have changed with that mechanism.

These are all private details of CPython that Nuitka integrates with, so I don't expect a pointer in the docs really. :)

**Comment by chwba** [comment]
I have tested the compile with python 3.8.7 and the latest release and got the same result. It works fine, the script starts.

**Comment by kayhayen** [comment]
With Nuitka stable, Python 3.9 support is still fresh, this is currently the only known incompatibility, but there may be unknown ones still @chwba 

I figured out, what this is about. In the C code of CPython it uses `PyType_Check` for which there is no Python equivalent anymore, because `isinstance` is saying that `list[int]` is a type, but `PyType_Check` does not. For Python 3.8 that kind of difference didn't happen and may be an issue of PEP 580 implementation.

There will have to be a new kind of node that makes this kind of type check and shall be used instead. That of course will also be an optimization. While at it, I would like to go over the Python reformulations and replace all `isinstance` uses with dedicated check node, they will be faster. The `isinstance` considers more things than CPython does, which risks compatibility obviously.

Lets see if the type check part can be released as part of a hotfix. Depending on how compex the solution is, I might push this back to 0.6.12 which is about ready too.

**Comment by kayhayen** [comment]
Checking things out, I noticed that for `PyType_Check` to be optimized in Python land, we need `issubclass`, but to use that we first need to know `isinstance` confirmed it as a type. So I ended up adding support for the built-in `issubclass`. This along with doing both checks in an `and` must be exactly what `PyType_Check` does.

Other type checks in Nuitka's reformulations of complex into simple Python are really instance checks, so there is nothing to be done there.

I split the change in 2 parts, one which will be in 0.6.11.5 once released, and the new optimization of the `issubclass` built-in calls to the new node. I am putting the full change through my CI now to see if there is something wrong with it. It's currently on the factory branch: https://nuitka.net/doc/factory.html
